lib/dpl/provider/pypi.rb

```diff
@@ -66,7 +66,9 @@ def pypi_wheel_arg
       end
 
       def install_deploy_dependencies
-        unless context.shell "wget -O - https://bootstrap.pypa.io/get-pip.py | python - --no-setuptools --no-wheel && " \
+        # --user likely fails inside virtualenvs but is needed outside to avoid needing sudo.
+        unless context.shell "if [ -z ${VIRTUAL_ENV+x} ]; then export PIP_USER=yes; fi && " \
```

Comment:
This creates hard to understand mix of languages. I'd defer env checks to Ruby side.

Reply:
@webknjaz I totally agree with you here. The only problem is that I do not master ruby at all, so if you give me an extra hint on how to do it, I would not mind doing it. Mainly I need to define this new envvar if the other one is defined.

Reply:
you can probably check `context.env['VIRTUAL_ENV']`

Reply:
What is `PIP_USER=yes` supposed to do?